### ISSUE TITLE
Hard gumbel softmax during training AIM?

### ISSUE THREAD

**Issue by tedhuang96** [limaosen0/AS-GCN#48]
I am confused with this line of code. If hard is set as True even for training, how can the gradient flow back through AIM module in backprop?

https://github.com/limaosen0/AS-GCN/blob/ab6951c3e6fc1b22bb4a0dc87a33cca228aa69fe/net/utils/adj_learn.py#L264
